packages/stylelint-plugin/src/rules/prefer-color-tokens.spec.ts

```diff
@@ -0,0 +1,89 @@
+// Migrated from https://github.com/cultureamp/node-packages/tree/0407d85/packages/stylelint-kaizen/lib
+// converted to TypeScript, and modified to be integratable with our stylelint plugin.
+
+import postcss from "postcss"
+import colorString from "color-string"
+import { defaultTheme } from "../../../design-tokens"
+import { messages, preferColorTokens } from "./prefer-color-tokens"
+
+function toRgb(color: string) {
+  return colorString.to.rgb(colorString.get(color)!.value)
+}
+
+const tests = {
+  accept: [
+    { code: ".foo { color: $color-blue-500; }" },
+    {
+      code: `
+        .foo {
+          z-index: 22;
+          padding: 1.5rem;
+          color: #123456;
+          background-color: rgb(123, 123, 123)
+        }
+    `,
+    },
+  ],
+  reject: [
+    {
+      code: `.foo { background-color: ${defaultTheme.color.blue[500]}; }`,
+      fixed: ".foo { background-color: $color-blue-500; }",
+      warnings: [
+        messages.expected("$color-blue-500", defaultTheme.color.blue[500]),
+      ],
+    },
+    {
+      code: `
+      .foo {
+        background-color: ${defaultTheme.color.blue[500]};
+        border-color: ${defaultTheme.color.purple[600]};
+      }`,
+      fixed: `
+      .foo {
+        background-color: $color-blue-500;
+        border-color: $color-purple-600;
+      }`,
+      warnings: [
+        messages.expected("$color-blue-500", defaultTheme.color.blue[500]),
+        messages.expected("$color-purple-600", defaultTheme.color.purple[600]),
+      ],
+    },
+  ],
+} as const
```

Comment:
These tests are inconsistent with the others in the project, however, i actually like this format the best out of all of them, and its the stylelint idiomatic way.

@jasononeil you've called out in previous work that we want to know which warnings exactly should be reported, rather than just a number of `expectedWarnings`, and this could be a good solution.

Reply:
FYI i've done something different in https://github.com/cultureamp/kaizen-design-system/pull/1836

I'm thinking the tests in this file should be changed to the style in #1836, with the function to do the testing split out to a shared testing lib file.

Reply:
Optionally, we could open another PR to make the testing style more consistent across the project.